consensus/hotstuff/timeoutaggregator/timeout_collectors.go

```diff
@@ -124,11 +134,15 @@ func (t *TimeoutCollectors) PruneUpToView(lowestRetainedView uint64) {
 	}
 	from := t.lowestRetainedView
 	t.lowestRetainedView = lowestRetainedView
+	numCollectors := len(t.collectors)
+	t.lock.Unlock()
+
+	t.metrics.TimeoutCollectorsRange(lowestRetainedView, t.newestViewCachedCollector, numCollectors)
```

Comment:
similarly here: lets make sure we capture the state atomically at the end of the update:
```suggestion
	newestViewCachedCollector := t.newestViewCachedCollector
	t.lock.Unlock()

	t.metrics.TimeoutCollectorsRange(lowestRetainedView, newestViewCachedCollector, numCollectors)
```